text/metrics/0001-naming-conventions.md

```diff
@@ -0,0 +1,149 @@
+# Metric naming conventions
+
+## Purpose
+
+Metric names and labels are the primary read-interface to metric data. The names and taxonomy need to be understandable and discoverable during routine exploration -- and this becomes critical during incidents.
+
+## Guidelines
+
+Namespace similar metrics together. Define top-level namespaces for common metric categories: for OS, like CPU and network; for app runtimes, like the JVM. This aids in discovery and adhoc comparison
+
+Provide consistent names for common label. Provides discoverability and disambiguation similar to metric names.
+
+Avoid semantic ambiguity. Use namespaced metric names in cases where similar metrics have significantly different implementations across the holistics of metrics. For example, every garbage collected runtime has a slightly different strategies and measures. Using common metric names for GC, not namespaces by the runtime, could create dissimilar comparisons and confusion for end users. Measures of operating system memory are similar.
+
+## Conventions
+
+All metrics have a limited set of common labels:
+* `system.host`
+
+### Operation System
+
+#### CPU
+
+`system.cpu` is the core metric name.
+
+Standard usage labels include, non-exhaustively,
+* `cpu.idle`
+* `cpu.user`
+* `cpu.sys`
+* `cpu.real`
+* `cpu.iowait`
+* `cpu.nice`
+* `cpu.interrupt`
+* `cpu.softirq`
+* `cpu.steal`
+
+A user can derive total CPU capacity by summing `system.cpu` across all labels.
+
+A user can derive CPU utilization by summing all values for the `cpu.idle` label and comparing that with all `system.cpu` values across all labels.
+
+`system.cpu` may include labels for per-core measures:
+* `cpu.core.[0-n]`, eg, `cpu.core.3`
+
+Cores should be reported ordinally as ordered by the operating system. It is recommended that values begin at 0.
+
+It is recommended that per-core labels should not be reported by default to reduce cardinality; a user should opt-in with via configuration.
+
+#### Network
+
+`system.network` is the core metric name.
+
+Standard labels include, non-exhaustively,
+* `network.sent`
+* `network.received`
+
+`system.network` may include labels for per-NIC measures:
+* `network.nic.[0-n]`, eg, `network.nic.3`
+
+NICs should be reported ordinally as ordered by the operating system. It is recommended that values begin at 0.
+
+Interfaces may also be reported by OS name, eg, `en3` in the label `network.nic.en3`.
+
+It is recommended that per-NIC labels should not be reported by default to reduce cardinality; a user should opt-in with via configuration.
+
+TODO: what other network labels? dropped? how about low-level things like window size?
+
+#### Memory
+
+`system.memory` is the core metric name.
+
+Standard labels include, non-exhaustively,
+* `memory.free`
+* `memory.resident`
+* `memory.shared`
+* `memory.private`
```

Comment:
This is a bad example because private and resident overlap; they shouldn't both be in the same metric.

Reply:
Yeah, actually I intended it that way because it introduces this complexity. I think we need to consider how we want to describe this in a non-overlapping way, and how we want to provide the more granular data elsewhere.

Reply:
I think every time there are two quantities that don't make sense to add together those quantities should be separate metrics; that means that, in this example, "resident", "shared", and "private" should all be separate metrics (I think it's possible for memory to be in two of the three categories simultaneously on Linux).

Reply:
Resolving now -- specifics about metric names and labels will be covered in a future doc

This is definitely a good point and I will circle back to you when it comes time to answer this question. It's a hard problem, I think.